.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+  security-events: none
+  statuses: read
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Or specify a tag to build from'
+        required: false
+        default: ''
+      confirm:
+        type: boolean
+        description: 'Yes, I''m sure'
+jobs:
+  release:
+    if: ${{ github.event.inputs.confirm == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check if user is in group"
+        run: |
+          MEMBER=$( curl -s \
+                 --request GET \
+                 --header 'authorization: Bearer ${{ secrets.GITTOKEN }}' \
+                 --header 'content-type: application/json' \
+                 --url https://api.github.com/orgs/matomo-org/teams/release-team/members \
+                 | jq '.[] | select(.login == "${{ github.actor }}")' )
+
+          if [[ -z "$MEMBER" ]]
+          then
+           echo "Action was not triggered by a member with release permission"
+           exit 1;
+          fi
+        shell: bash
+      - uses: actions/checkout@v2
+        with:
+          lfs: false
+      - name: Import GPG key
+        id: import_gpg
+        run: |
+          echo "${{ secrets.GPG_CERTIFICATE }}" > $HOME/private.asc
+          gpg --import --batch --yes $HOME/private.asc
+          echo "default-cache-ttl 7200
+          max-cache-ttl 31536000
+          allow-preset-passphrase" > $HOME/.gnupg/gpg-agent.conf
+          keygrip=$(gpg --import --import-options show-only --with-keygrip $HOME/private.asc | grep "Keygrip" | grep -oP "([A-F0-9]+)" | head -1)
+          hexPassphrase=$( echo -n "${{ secrets.GPG_CERTIFICATE_PASS }}" | od -A n -t x1 -w100 | sed 's/ *//g' )
+          gpg-connect-agent "RELOADAGENT" /bye
+          gpg-connect-agent "PRESET_PASSPHRASE ${keygrip} -1 ${hexPassphrase}" /bye
+          gpg-connect-agent "KEYINFO ${keygrip}" /bye
+      - name: Check preconditions, create tag, build and publish release
+        id: tag
+        run: |
+          if [[ -n "${{ github.event.inputs.version }}" ]]
+          then
+            version="${{ github.event.inputs.version }}"
+            echo "Version to re-build: '$version'"
+
+            git fetch --tags -q 2>/dev/null
+            tag_exists=$( git tag --list "$version" )
+
+            if [[ -z "$tag_exists" ]]
+            then
+              echo "A tag for $version does not exist."
+              exit 1
+            fi
+
+            echo ::set-output name=update::true
+          else
+            version=$( cat core/Version.php | grep -oP "VERSION = '\K([^\']+)" )
+            echo "Version to build: '$version'"
+
+            git fetch --tags -q 2>/dev/null
+            tag_exists=$( git tag --list "$version" )
+
+            if [[ -n "$tag_exists" ]]
+            then
+              echo "A tag for $tag_exists already exists."
+              exit 1
+            fi
+
+            if [[ ${GITHUB_REF#refs/heads/} != "4.x-dev" && ${GITHUB_REF#refs/heads/} != "next_release" ]]
```

Comment:
Can we update the docs in https://developer.matomo.org/guides/release-management#when-starting-to-work-on-the-next-major-release that we need to update this release workflow when starting to work on a new major release? I assume we will need to change then `4.x-dev` -> `5.x-dev`

Reply:
changed the check. It can now be any branch that matches `[4-9]\.x-dev`. So that should work for quite a while...